packages/@uppy/aws-s3-multipart/src/createSignedURL.js

```diff
@@ -90,32 +114,32 @@ export default async function createSignedURL ({
 }) {
   const Service = 's3'
   const host = `${bucketName}.${Service}.${Region}.amazonaws.com`
-  const CanonicalUri = `/${encodeURI(Key)}`
+  const CanonicalUri = `/${Key.split('/').map(extendedEncodeURIComponent).join('/')}`
```

Comment:
List of char is taken from https://tc39.es/ecma262/#sec-encodeuri-uri
```suggestion
  const CanonicalUri = `/${encodeURI(Key).replace(/[;?:@&=+$,#!'()*]/g, percentEncode)}`
```

Reply:
https://github.com/transloadit/uppy/pull/5006/commits/3ab6024bfca48658a7fea4a4e202c3033a22d5a8